---
title: "RNA_Seq_AnalysisCalu3_Mock_Cov2"
output:
  html_document: default
  pdf_document: default
editor_options: 
  chunk_output_type: console
---

#Setup

First we need to load all sorts of packages that enable us to analyze a read count table that we generated using the nextflow nf-core pipeline version 1.4.3 on a fused genome build of human and SARS-Cov2.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tximportData)
library(ggrepel)
library(DESeq2)
library(tidyverse)
library(patchwork)
library(ggrastr)
library(fgsea)

```


## B110 Theme

```{r theme, include=FALSE}

theme_b110 <- function(){
  theme_classic() +
  theme(
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 10),
    plot.title = element_text(size = 12,hjust = 0.5,face = "bold"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10),
    legend.position = "bottom"
    )
}

```

## B110 Colors

```{r colors}

sgi_blue    = '#5087C8'
sgi_yellow1 = '#F2EE35'
sgi_yellow2 = '#FED98E'
b110_grey   = '#808080'
b110_grey_light   = '#909090'
b110_transparent_black = alpha('#000000',0.5)
google_red = '#dd4b39'
google_green = '#0F9D58'
google_yellow = '#F4B400'
google_blue = '#4285F4'

```

## Read in the meta data

```{r meta_data}

raw_meta_data <- read_delim("../meta_data/19361_meta.tsv",delim = "\t")

raw_meta_data %<>% 
  filter(grepl("R1",FASTQ_FILE)) %>% 
  mutate(run = gsub(pattern = "(.+)_.*.fastq.gz",replacement = "\\1",x = FASTQ_FILE)) %>% 
  dplyr::rename(sample = SAMPLE_NAME)


```

#import the data from star

```{r,warning=F,message=F}

#require(tximport)
#txi <- tximport(files, type = "salmon", tx2gene = tx2gene,ignoreTxVersion = TRUE)
#names(txi)

star_data <- read_delim("../data/merged_gene_counts.txt",delim = "\t")

colnames(star_data) <- gsub(pattern = "(\\S+?)Aligned.sortedByCoord.out.bam",replacement = "\\1",x = colnames(star_data))

star_data_id2name <- star_data %>% select(-starts_with("AS"))

star_data %<>% 
  select(-gene_name) %>% 
  column_to_rownames("Geneid") %>% 
  as.data.frame()
```

Because of the nature of the experiment a lot of reads fall onto the viral genome of SARS-Cov2. Thus, we check the percentage of SARS RNA reads in each sample.

```{r}

plot_data <- 
  star_data %>% 
  rownames_to_column("gene_id") %>% 
  gather(sample,reads,-gene_id) %>% 
  mutate(viral_read = grepl("ENSSAS",gene_id)) %>% 
  group_by(sample,viral_read) %>% 
  summarise(read_count=sum(reads)) %>% 
  rename(run=sample) %>% 
  left_join(vibhu_anno) %>%
  mutate(sample=factor(sample,levels = c(paste0("V",1:16))))

a <-  plot_data%>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="stack" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))

b <- plot_data %>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="stack" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))+ facet_wrap(~viral_read)

c <- plot_data %>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="fill" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))

a + b + c + plot_layout(nrow = 2,ncol = 2)

ggsave("../plots/per_cent_viral_reads.pdf",a + b + c + plot_layout(nrow = 2,ncol = 2),width = 16,height = 12)
```



#prepare the DESeq2 data object

```{r,warning=F,message=F}

#from the raw annotation file we need to extract the treatment and time conditions 

# in theory this could add power to the differemt block countin similar adjacent time points as replicates if this is needed

vibhu_anno <- read_delim("../meta_data/vbhu_annotations.txt",delim = "\t")

vibhu_anno %<>% 
    mutate(virus=(if_else(grepl("Cov|CoV",treatment),"Cov2","Mock")),
         drug=(if_else(grepl("8C",treatment)&!grepl("^0µM",concentration),"4µ8C",
                             if_else(grepl("DTT",treatment),"DTT",NA_character_))),
         concentration=(if_else(grepl("^0µM",concentration),NA_character_,concentration))) %>%
  mutate(sample=paste0("V",sample)) %>%
  left_join(raw_meta_data %>% dplyr::select(sample,run)) %>%
  select(sample,run,virus,drug,concentration) 

```

## Analysis for DTT testing

# design table preparation

```{r}

sampleTable_DTT_experiment <- vibhu_anno %>% 
                                filter(is.na(concentration)) %>%
                                mutate(virus=factor(if_else(is.na(drug),virus,"DTT"))) %>%
                                select(sample,run,virus) %>% 
                                as_data_frame()

rownames(sampleTable_DTT_experiment)<-sampleTable_DTT_experiment$run

```


#simple prefiltering

```{r,warning=F,message=F}
star_data_DTT_experiment <- star_data[,rownames(sampleTable_DTT_experiment)]

dds_DTT_experiment <- DESeqDataSetFromMatrix(star_data_DTT_experiment, sampleTable_DTT_experiment, ~virus)

#counting transcripts

nrow(dds_DTT_experiment)

#excluding genes that were not matched at all
dds_DTT_experiment <- dds_DTT_experiment[ rowSums(counts(dds_DTT_experiment)) > 1, ]
nrow(dds_DTT_experiment)

#remaining 28960

```

#stabilize variance for low readcount genes

```{r,warning=F,message=F}

vsd <- vst(dds_DTT_experiment, blind = FALSE)

```

#cluster data to get an overview

Treatments should cluster together

```{r,warning=F,message=F}
require("pheatmap")
require("RColorBrewer")

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- vsd$sample
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
annotations = data.frame("virus"=vsd$virus)
row.names(annotations)<-vsd$sample

pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,annotation_row = annotations)


# --> nice it clusters the gene expression profile by age # 2nd analysis: now it clusters nicely in sample but horribly between batches


c <- plotPCA(vsd, "virus") + ggtitle("PCA by virus")

ggsave(c,filename = "../plots/PCA_pre_batch_correct_dds_DTT_experiment.pdf")

```

# lets call differential genes

```{r,warning=F,message=F}

dds <- DESeq(dds_DTT_experiment)

# get available contrasts

resultsNames(dds)

# choose the contrast which shows the difference between treatments given background differences of the treatment response over time.

DTT_over_Mock <- results(dds,contrast = c("virus","DTT","Mock"))

Cov2_over_Mock <- results(dds,contrast = c("virus","Cov2","Mock"))

summary(DTT_over_Mock)

summary(Cov2_over_Mock)
#res %>% as.data.frame() %>% View()
# --> there is quite a number of genes differential expressed --> lets look at who they are
```



# Cov2 vs DTT

```{r,warning=F,message=F}

plotCounts(dds, gene = "ENSG00000126368", intgroup = c("virus"),returnData = F) 

# here we threshold the data simply in upregulated and down regulated according to foldchange and sort by FDR
# for enrichment of functional annotation clustering it make further sense to threshold by fold change and FDR

DTT_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>%
  write_delim(path = "../processed_data/star_top_table_DTT_over_Mock_full.txt",delim = "\t")

down_regulated_genes_DTT <- DTT_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange<0,padj<0.1)
  
write_delim(down_regulated_genes_DTT,"../processed_data/star_top_table_DTT_over_Mock_downDTT.txt",delim = "\t")

up_regulated_genes_DTT <- DTT_over_Mock %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange>0,padj<0.1)

write_delim(up_regulated_genes_DTT,"../processed_data/star_top_table_DTT_over_Mock_upDTT.txt",delim = "\t")

Cov2_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>%
  write_delim(path = "../processed_data/star_top_table_Cov2_over_Mock_full.txt",delim = "\t")

down_regulated_genes_Cov2 <- Cov2_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange<0,padj<0.1)
  
write_delim(down_regulated_genes_Cov2,"../processed_data/star_top_table_Cov2_over_Mock_downCov2.txt",delim = "\t")

up_regulated_genes_Cov2 <- Cov2_over_Mock %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange>0,padj<0.1)

write_delim(up_regulated_genes_Cov2,"../processed_data/star_top_table_Cov2_over_Mock_upCov2.txt",delim = "\t")
```

##Vizual inspection

#DTT_over_Mock

```{r}
# next we visualize the differential expressed genes using MA plot of base line expression versus fold change and foldchange versus -log10 pvalue
# here all genes that are deferentially expressed between SARS-Cov2 infection and Mock treatment with an FDR < 0.1 are colored blue.
# the top 15 differentially expressed genes according the the FDR ranking are labeled by their official gene symbol

this_data <- DTT_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na()

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "../plots/star_differential_expressed_volcano_DTT_over_Mock.pdf")
ggsave(p2,filename = "../plots/star_differential_expressed_MA_DTT_over_Mock.pdf")

# next we visualize the same data without the viral genes


this_data <- DTT_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na() %>%
  filter(grepl("ENSG",Geneid))

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "../plots/star_differential_expressed_volcano_DTT_over_Mock_woCov2.pdf")
ggsave(p2,filename = "../plots/star_differential_expressed_MA_DTT_over_Mock_woCov2.pdf")
```

#DTT_over_Mock

```{r}
# next we visualize the differential expressed genes using MA plot of base line expression versus fold change and foldchange versus -log10 pvalue
# here all genes that are deferentially expressed between SARS-Cov2 infection and Mock treatment with an FDR < 0.1 are colored blue.
# the top 15 differentially expressed genes according the the FDR ranking are labeled by their official gene symbol

this_data <- Cov2_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na()

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
            geom_label_repel(data = this_data %>% 
                               filter(significant=="significant") %>% 
                               arrange(padj) %>% 
                               head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% 
                        filter(significant=="significant") %>% 
                        arrange(padj) %>% 
                        head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "../plots/star_differential_expressed_volcano_Cov2_over_Mock.pdf")
ggsave(p2,filename = "../plots/star_differential_expressed_MA_Cov2_over_Mock.pdf")

# next we visualize the same data without the viral genes


this_data <- Cov2_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na() %>%
  filter(grepl("ENSG",Geneid))

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% 
                              filter(significant=="significant") %>% 
                              arrange(padj) %>% 
                              head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% 
                        filter(significant=="significant") %>% 
                        arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "../plots/star_differential_expressed_volcano_Cov2_over_Mock_woCov2.pdf")
ggsave(p2,filename = "../plots/star_differential_expressed_MA_Cov2_over_Mock_woCov2.pdf")
```

#Interpretation

Gene set annotations are downloaded from MSigDB
https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp#H
29.05.2020: 14:42 CET
C5 : GO Bioprocess

C7 : Immunology

h.all : Hallmark gene sets

Citation for hallmark collection:

Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi M, Mesirov JP, Tamayo P. The Molecular Signatures Database (MSigDB) hallmark gene set collection. Cell Syst. 2015 Dec 23;1(6):417-425. 
 

##Signature enrichment

#DTT_over_Mock

```{r}

# Load the pathways into a named list
pathways.hallmark <- gmtPathways("../meta_data/h.all.v7.1.symbols.gmt")

pathways.go <- gmtPathways("../meta_data/c5.bp.v7.1.symbols.gmt")

this_data <- DTT_over_Mock %>% as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)

seq_results <- this_data %>%
  select(stat=stat,Symbol=gene_name,Geneid) %>%
  column_to_rownames("Geneid") %>%
  na.omit() %>% 
  distinct() %>% 
  group_by(Symbol) %>% 
  summarize(stat=mean(stat)) %>%
  ungroup() %>%
  deframe()

#GSEA of hallmark sets

fgsea_hallmark <- fgsea(pathways=pathways.hallmark, stats=seq_results)
fgsea_hallmarktdy <- fgsea_hallmark %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_hallmarktdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_b110()+
  scale_fill_manual(values = c(b110_grey,google_blue))

ggsave("../plots/star_enriched_signatures_DTT_over_Mock.pdf")

#GSEA of GO-term sets

fgsea_go <- fgsea(pathways=pathways.go, stats=seq_results)
fgsea_gotdy <- fgsea_go %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_gotdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

ggsave("../plots/star_enriched_GO_signatures_DTT_over_Mock.pdf")
```


### GSEA analysis

We perform gene set enrichment analysis using the Broad Institute's [GSEA](http://software.broadinstitute.org/gsea/index.jsp) [@pmid17644558]. An R version of the algorithm is implemented in the `fgsea` algorithm [@Sergushichev060012], which we use for analysis.

We want to visualize the results as a barcode plot. `fgsea` already implements a nice barcode plot, which we cusotomize a bit to adapt it according to our expectations.

Code for the abrcode plot was inspired by [@pmid31097693] . c Benedikt Rauscher

```{r, results='hide', warning=F, message=F}
custom_barcode_plot <- function(df, sig){
  ## named vector of gene-level stats
  stat_vector <- setNames(df$stat, df$gene_name)
  ## genes in signature
  sig_genes <- pathways.hallmark[[sig]]
  
  ## generate barcode plot
  bc_plot <- plotEnrichment(sig_genes, stat_vector)
  
  ## remove unwanted layers
  bc_plot$layers <- list()
  
  ## add barcode at the bottom
  lowest_pos <- min(bc_plot$data[,2])
  dash_length <- abs(reduce(range(bc_plot$data[,2]), `-`)*0.1)
  middle <- which.min(abs(sort(df$stat, decreasing=T)))
  
  bc_plot_custom <- bc_plot + geom_segment(aes(x=x, xend=x), y=lowest_pos,
                           yend=lowest_pos-dash_length) + 
    geom_line(colour='#4daf4a') + 
    geom_hline(yintercept=lowest_pos, colour='#cccccc') + 
    geom_hline(yintercept=0, colour='#cccccc') + xlab('') +
    theme_classic() +
    geom_tile(data=tibble(rank=1:length(stat_vector), 
                          y=lowest_pos-(1.25*dash_length)), 
              aes(x=rank, y=y, fill=rank),
                  width=1,
                  height=0.5*dash_length) +
    scale_fill_gradient2(low ='#b2182b', high='#2166ac', 
                         mid='#f7f7f7', midpoint = middle) + 
    scale_x_continuous(expand = c(0, 0)) +
    scale_y_continuous(expand = c(0, 0)) +
    theme(panel.grid=element_blank(), 
          axis.text.x=element_blank(),
          axis.ticks.x = element_blank(),
          legend.position = 'none') + 
    ggtitle(paste(sig, 'signature')) +
    ylab('Enrichment score')
  
  return(bc_plot_custom)
}
```

barcode plots

```{r}

down_fgsea <- fgsea_hallmark %>% arrange(NES)

## generate plots
bc_plots_down <- map(1:8, function(j){#nrow()
  bcp <- custom_barcode_plot(this_data, down_fgsea$pathway[j]) + 
    annotate('text', x=Inf , y=Inf, hjust=1, vjust=1, 
             label=paste('NES =', round(down_fgsea$NES[j], 2), 
                         '\nFDR =', round(down_fgsea$padj[j], 3)))
  
  return(bcp)
})

up_fgsea <- fgsea_hallmark %>% arrange(-NES)

## generate plots
bc_plots_up <- map(1:8, function(j){#nrow()
  bcp <- custom_barcode_plot(this_data, up_fgsea$pathway[j]) + 
    annotate('text', x=Inf , y=Inf, hjust=1, vjust=1, 
             label=paste('NES =', round(up_fgsea$NES[j], 2), 
                         '\nFDR =', round(up_fgsea$padj[j], 3)))
  
  return(bcp)
})

## plot to canvas
reduce(c(bc_plots_up), `+`) + plot_layout(ncol=2)

  ggsave("../plots/star_signature_enrichment_upregulated_DTT_over_Mock.pdf")

## plot to canvas
reduce(c(bc_plots_down), `+`) + plot_layout(ncol=2)

ggsave("../plots/star_signature_enrichment_downregulated_DTT_over_Mock.pdf")


```

#Cov2_over_Mock

```{r}

# Load the pathways into a named list
pathways.hallmark <- gmtPathways("../meta_data/h.all.v7.1.symbols.gmt")

pathways.go <- gmtPathways("../meta_data/c5.bp.v7.1.symbols.gmt")

this_data <- Cov2_over_Mock %>% as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)

seq_results <- this_data %>%
  select(stat=stat,Symbol=gene_name,Geneid) %>%
  column_to_rownames("Geneid") %>%
  na.omit() %>% 
  distinct() %>% 
  group_by(Symbol) %>% 
  summarize(stat=mean(stat)) %>%
  ungroup() %>%
  deframe()

#GSEA of hallmark sets

fgsea_hallmark <- fgsea(pathways=pathways.hallmark, stats=seq_results)
fgsea_hallmarktdy <- fgsea_hallmark %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_hallmarktdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_b110()+
  scale_fill_manual(values = c(b110_grey,google_blue))

ggsave("../plots/star_enriched_signatures_Cov2_over_Mock.pdf")

#GSEA of GO-term sets

fgsea_go <- fgsea(pathways=pathways.go, stats=seq_results)
fgsea_gotdy <- fgsea_go %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_gotdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

ggsave("../plots/star_enriched_GO_signatures_Cov2_over_Mock.pdf")
```


## Example collection

Next we plot the top differential expressed genes as a function of time and infection

```{r}

LUT <- star_data_id2name %>% column_to_rownames("Geneid")

topGenes_up <- DTT_over_Mock %>%
  as.data.frame() %>%
  arrange(desc(stat)) %>%
  rownames_to_column("name") %>%
  filter(log2FoldChange>0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_up <- map(topGenes_up, function(topGene){#nrow()
   geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("virus"),returnData = TRUE) 
   ggplot(geneCounts, aes(x = virus, y = count)) +
              scale_y_log10() +  
              geom_point(size = 3) + 
              geom_smooth() +
              theme_b110() +
              scale_color_manual(values = c(google_red,b110_grey)) + 
              ylab("norm. count") +
              xlab("time [h]") +
              ggtitle(LUT[topGene,1])
})

topGenes_down <- DTT_over_Mock %>%
  as.data.frame() %>%
  arrange(stat) %>%
  rownames_to_column("name") %>%
  filter(log2FoldChange<0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_down <- map(topGenes_down, function(topGene){#nrow()
  geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("virus"),returnData = TRUE) 
   ggplot(geneCounts, aes(x = virus, y = count)) +
            scale_y_log10() +  
            geom_point(size = 3) + 
            geom_smooth() +
            theme_b110() +
            scale_color_manual(values = c(google_red,b110_grey)) + 
            ylab("norm. count") +
            xlab("time [h]") +
            ggtitle(LUT[topGene,1])
})

## plot to canvas
reduce(c(count_plots_up), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave("../plots/star_top_up-regulated_genes_DTT_over_Mock.pdf")

## plot to canvas
reduce(c(count_plots_down), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave("../plots/star_top_down-regulated_genes_DTT_over_Mock.pdf")

```

# DTT DEG vs Cov2 DEG

```{r}

DTT_DEG <- DTT_over_Mock %>% 
   as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)%>% 
  filter(padj<0.1,grepl("ENSG",Geneid))


Cov2_DEG <- Cov2_over_Mock %>% 
   as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name) %>% 
  filter(padj<0.1,grepl("ENSG",Geneid))

joined_DEG <- DTT_DEG %>%
  full_join(Cov2_DEG,suffix = c("DTT","Cov2"),by = c("Geneid","gene_name")) %>%
  as_tibble()

joined_DEG %>% drop_na() %>% ggplot(aes(x=log2FoldChangeDTT,y=log2FoldChangeCov2)) + geom_point() + theme_b110() + geom_vline(xintercept = 0) + geom_hline(yintercept = 0)

joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2>0,log2FoldChangeDTT>0) %>%
  write_delim("../processed_data/Cov2_shared_DTT_positive_DEG.txt",delim = "\t")

seq_results <- joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2>0,log2FoldChangeDTT>0) %>% 
  select(stat=statCov2,Symbol=gene_name,Geneid) %>%
  column_to_rownames("Geneid") %>%
  na.omit() %>% 
  distinct() %>% 
  group_by(Symbol) %>% 
  summarize(stat=mean(stat)) %>%
  ungroup() %>%
  deframe()

fgsea_go <- fgsea(pathways=pathways.go, stats=seq_results,scoreType = "pos")
fgsea_gotdy <- fgsea_go %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_gotdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

ggsave("../plots/star_enriched_GO_signatures_Cov2_shared_DTT_positive.pdf")

joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2<0,log2FoldChangeDTT<0) %>%
  write_delim("../processed_data/Cov2_shared_DTT_negative_DEG.txt",delim = "\t")


seq_results <- joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2<0,log2FoldChangeDTT<0) %>% 
  select(stat=statCov2,Symbol=gene_name,Geneid) %>%
  column_to_rownames("Geneid") %>%
  na.omit() %>% 
  distinct() %>% 
  group_by(Symbol) %>% 
  summarize(stat=mean(stat)) %>%
  ungroup() %>%
  deframe()

fgsea_go <- fgsea(pathways=pathways.go, stats=seq_results,scoreType = "neg")
fgsea_gotdy <- fgsea_go %>%
  as_tibble() %>%
  arrange(desc(NES))

fgsea_gotdy %>% 
  arrange(padj) %>%
  head(.,40) %>%
  ggplot(aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

ggsave("../plots/star_enriched_GO_signatures_Cov2_shared_DTT_negative.pdf")


p1 <- euler(list("DTT" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeDTT),log2FoldChangeDTT<0) %>% 
  pull(gene_name),
  "Cov2" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeCov2),log2FoldChangeCov2<0) %>% 
  pull(gene_name)))

plot(p1,main="down regulated genes")

pdf("../plots/DTT_Cov2_shared_downregulated_genes.pdf")
plot(p1,main="down regulated genes")
dev.off()

p2 <- euler(list("DTT" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeDTT),log2FoldChangeDTT>0) %>% 
  pull(gene_name),
  "Cov2" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeCov2),log2FoldChangeCov2>0) %>% 
  pull(gene_name)))

plot(p2,main="up regulated genes")
pdf("../plots/DTT_Cov2_shared_upregulated_genes.pdf")
plot(p2,main="up regulated genes")
dev.off()
```


# write table of detected genes

```{r}

tibble("Geneid"=rownames(res)) %>% left_join(star_data_id2name) %>% write_delim("../processed_data/detected_genes_DTT_experiment.txt",delim = "\t")

```

# Session info

```{r}
writeLines(capture.output(sessionInfo()), "../processed_data/SessionInfo_DTT_experiment.txt")
```







