---
title: "RNA_Seq_AnalysisCalu3_Mock_Cov2"
output:
  html_document: default
  pdf_document: default
editor_options: 
  chunk_output_type: console
---

#Setup

First we need to load all sorts of packages that enable us to analyze a read count table that we generated using the nextflow nf-core pipeline version 1.4.3 on a fused genome build of human and SARS-Cov2.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tximportData)
library(ggrepel)
library(DESeq2)
library(tidyverse)
library(patchwork)
library(ggrastr)
library(fgsea)

```


## B110 Theme

```{r theme, include=FALSE}

theme_b110 <- function(){
  theme_classic() +
  theme(
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 10),
    plot.title = element_text(size = 12,hjust = 0.5,face = "bold"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10),
    legend.position = "bottom"
    )
}

```

## B110 Colors

```{r colors}

sgi_blue    = '#5087C8'
sgi_yellow1 = '#F2EE35'
sgi_yellow2 = '#FED98E'
b110_grey   = '#808080'
b110_grey_light   = '#909090'
b110_transparent_black = alpha('#000000',0.5)
google_red = '#dd4b39'
google_green = '#0F9D58'
google_yellow = '#F4B400'
google_blue = '#4285F4'

```

## Read in the meta data

```{r meta_data}

raw_meta_data <- read_delim("meta_data/31983_meta.tsv",delim = "\t")

raw_meta_data %<>% 
  filter(grepl("R1",FASTQ_FILE)) %>% 
  mutate(
    run = gsub(pattern = "(.+)_.*.fastq.gz",
               replacement = "\\1",x = FASTQ_FILE)) %>% 
  dplyr::rename(sample = SAMPLE_NAME)


```

#import the data from star

```{r,warning=F,message=F}

star_data <- read_delim("data/merged_gene_counts.txt",delim = "\t")

colnames(star_data) <- gsub(pattern = "(\\S+?)Aligned.sortedByCoord.out.bam",replacement = "\\1",x = colnames(star_data))

star_data_id2name <- star_data %>% select(-starts_with("AS"))


star_data %<>%
  gather(run,count,-Geneid,-gene_name) %>% 
  separate(run,c("AS","number1","LR","number2")) %>%
  group_by(Geneid,gene_name,AS,number1) %>%
  summarise(count=sum(count)) %>%
  ungroup() %>% 
  select(Geneid,gene_name,number1,count) %>%
  spread(number1,count)

star_data %<>% 
  select(-gene_name) %>% 
  column_to_rownames("Geneid") %>% 
  as.data.frame()

#star_data <- star_data[-grep("ENSSAS",rownames(star_data)),]

```


#prepare the DESeq2 data object

```{r,warning=F,message=F}

#from the raw annotation file we need to extract the treatment and time conditions 

# in theory this could add power to the differemt block countin similar adjacent time points as replicates if this is needed

raw_meta_data <-
    raw_meta_data %>% separate(sample,c("number","infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  filter(treatment!="4µ8C") %>%
   filter(treatment!="4u8C") %>%
  mutate(treatment=if_else(treatment!="NonInf","Kira",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()
  

```

#lets check the percent of viral reads per sample

Because of the nature of the experiment a lot of reads fall onto the viral genome of SARS-Cov2. Thus, we check the percentage of SARS RNA reads in each sample.

```{r}

plot_data <- 
  star_data %>% 
  rownames_to_column("gene_id") %>% 
  gather(sample,reads,-gene_id) %>% 
  mutate(viral_read = grepl("ENSSAS",gene_id)) %>% 
  group_by(sample,viral_read) %>% 
  summarise(read_count=sum(reads)) %>% 
  rename(run=sample) %>% 
  left_join(raw_meta_data) %>%
  arrange(infection,treatment) %>%
  distinct() %>% drop_na()

a <-  plot_data %>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="stack" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))

b <- plot_data %>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="stack" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))+ facet_wrap(~viral_read)

c <- plot_data %>% ggplot(aes(x=sample,y=read_count,fill=viral_read)) + geom_bar(stat = "identity",position="fill" ) + theme_b110() +scale_fill_manual(values = c(b110_grey,google_red)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=0.9))

a + b + c + plot_layout(nrow = 2,ncol = 2)

ggsave("plots/per_cent_viral_reads.pdf",a + b + c + plot_layout(nrow = 2,ncol = 2),width = 16,height = 12)
```



## Analysis for kira testing

# design table preparation

```{r}

sampleTable_kira_experiment <- 
  raw_meta_data %>% 
    filter(concentration %in% c("2uM","1uM","0uM"),treatment %in% c("Kira","DMSO"))  %>%
    mutate(treatment=if_else(treatment=="DMSO","Kira",treatment)) %>%
    mutate(infection=factor(infection)) %>% 
    as_data_frame() %>% distinct() %>% 
    filter(infection=="MOI10")

rownames(sampleTable_kira_experiment)<-sampleTable_kira_experiment$run

```


#simple prefiltering

```{r,warning=F,message=F}

star_data_kira_experiment <- 
  star_data[,
            rownames(sampleTable_kira_experiment )
            ]

dds_kira_experiment <- DESeqDataSetFromMatrix(star_data_kira_experiment, sampleTable_kira_experiment, ~1)

#counting transcripts

nrow(dds_kira_experiment)

#excluding genes that were not matched at all
dds_kira_experiment <- dds_kira_experiment[ rowSums(counts(dds_kira_experiment)) > 1, ]
nrow(dds_kira_experiment)

#remaining 27598

```

#Normalized Viral counts

```{r}

star_data_kira_experiment <- 
  star_data_kira_experiment %>% 
    filter(rowSums(.)>10) %>% 
    mutate(across(everything(),~.x+1),across(everything(),~log(.x/median(.x)))) %>%
    rownames_to_column("Geneid" ) %>%
    left_join(star_data_id2name)

star_data_kira_experiment[grepl("ENSSAS",rownames(star_data_kira_experiment)),]

a <- star_data_kira_experiment %>% 
  mutate(across(one_of(c('865904','865905','865901')), ~.x - `865901`)) %>%
  drop_na() %>%
  arrange(desc(abs(`865905`))) %>%
  filter(grepl("ENSSAS",Geneid)) %>%
  gather(run,value,-Geneid,-gene_name) %>% 
  left_join(raw_meta_data) %>%
  mutate(concentration=parse_number(concentration)) %>%
  ggplot(aes(x=concentration,y=value,group=Geneid)) +
    geom_line(linetype = "dashed") +
    geom_point()

b <- star_data_kira_experiment %>% 
  mutate(across(one_of(c('865904','865905','865901')), ~.x - `865901`)) %>%
  drop_na() %>%
  arrange(desc(abs(`865905`))) %>%
  filter(grepl("ENSSAS",Geneid)) %>%
  gather(run,value,-Geneid,-gene_name) %>% 
  left_join(raw_meta_data) %>%
  #mutate(concentration=parse_number(concentration)) %>%
  ggplot(aes(x=concentration,y=gene_name,fill=value)) +
    geom_tile() +
    scale_fill_gradient2()

print(a+b)

ggsave("plots/Kira_experiment_Cov2counts.pdf",plot=a+b)
```

#stabilize variance for low readcount genes

```{r,warning=F,message=F}

vsd <- vst(dds_kira_experiment, blind = FALSE)

```

#cluster data to get an overview

Treatments should cluster together

```{r,warning=F,message=F}
require("pheatmap")
require("RColorBrewer")

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- vsd$sample
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
annotations = data.frame("virus"=vsd$infection)
row.names(annotations)<-vsd$sample

pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,annotation_row = annotations)

c <- plotPCA(vsd, c("infection")) + ggtitle("PCA by virus") + theme_b110()
d <- plotPCA(vsd, c("concentration")) + ggtitle("PCA by conc")+ theme_b110()

print(c+d)
ggsave(c+d,filename = "plots/PCA_pre_batch_correct_dds_kira_experiment_wCov2.pdf")

```

# lets call differential genes

```{r,warning=F,message=F}

dds <- DESeq(dds_kira_experiment)

# get available contrasts

resultsNames(dds)

# choose the contrast which shows the difference between treatments given background differences of the treatment response over time.

xkira_over_Mock <- results(dds,contrast = c("concentration","2uM","0uM"))

Cov2_over_Mock <- results(dds,contrast = c("infection","NonInf","MOI10"))

summary(results(dds))


summary(Cov2_over_Mock)
#res %>% as.data.frame() %>% View()
# --> there is quite a number of genes differential expressed --> lets look at who they are
```



# Cov2 vs kira

```{r,warning=F,message=F}

plotCounts(dds, gene = "ENSG00000051523", intgroup = c("infection"),returnData = F) 

# here we threshold the data simply in upregulated and down regulated according to foldchange and sort by FDR
# for enrichment of functional annotation clustering it make further sense to threshold by fold change and FDR

xkira_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>%
  arrange(padj) %>%
  write_delim(path = "processed_data/star_top_table_xkira_over_Mock_full.txt",delim = "\t")

down_regulated_genes_kira <- xkira_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange<0,padj<0.1)
  
write_delim(down_regulated_genes_kira,"processed_data/star_top_table_xkira_over_Mock_downkira.txt",delim = "\t")

up_regulated_genes_kira <- xkira_over_Mock %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange>0,padj<0.1)

write_delim(up_regulated_genes_kira,"processed_data/star_top_table_xkira_over_Mock_upkira.txt",delim = "\t")

Cov2_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>%
  write_delim(path = "processed_data/star_top_table_kira_Cov2_over_Mock_full.txt",delim = "\t")

down_regulated_genes_Cov2 <- Cov2_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange<0,padj<0.1)
  
write_delim(down_regulated_genes_Cov2,"processed_data/star_top_table_kira_Cov2_over_Mock_downCov2.txt",delim = "\t")

up_regulated_genes_Cov2 <- Cov2_over_Mock %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange>0,padj<0.1)

write_delim(up_regulated_genes_Cov2,"processed_data/star_top_table_kira_Cov2_over_Mock_upCov2.txt",delim = "\t")
```

##Vizual inspection

#xkira_over_Mock

```{r}
# next we visualize the differential expressed genes using MA plot of base line expression versus fold change and foldchange versus -log10 pvalue
# here all genes that are deferentially expressed between SARS-Cov2 infection and Mock treatment with an FDR < 0.1 are colored blue.
# the top 15 differentially expressed genes according the the FDR ranking are labeled by their official gene symbol

this_data <- xkira_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na()

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_xkira_over_Mock.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_xkira_over_Mock.pdf")

# next we visualize the same data without the viral genes


this_data <- xkira_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na() %>%
  filter(grepl("ENSG",Geneid))

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_xkira_over_Mock_woCov2.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_xkira_over_Mock_woCov2.pdf")
```

#Cov2_over_Mock

```{r}
# next we visualize the differential expressed genes using MA plot of base line expression versus fold change and foldchange versus -log10 pvalue
# here all genes that are deferentially expressed between SARS-Cov2 infection and Mock treatment with an FDR < 0.1 are colored blue.
# the top 15 differentially expressed genes according the the FDR ranking are labeled by their official gene symbol

this_data <- Cov2_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na()

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
            geom_label_repel(data = this_data %>% 
                               filter(significant=="significant") %>% 
                               arrange(padj) %>% 
                               head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% 
                        filter(significant=="significant") %>% 
                        arrange(padj) %>% 
                        head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_Cov2_over_Mock.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_Cov2_over_Mock.pdf")

# next we visualize the same data without the viral genes


this_data <- Cov2_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na() %>%
  filter(grepl("ENSG",Geneid))

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point_rast(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point_rast(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% 
                              filter(significant=="significant") %>% 
                              arrange(padj) %>% 
                              head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point_rast(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point_rast(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% 
                        filter(significant=="significant") %>% 
                        arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_Cov2_over_Mock_woCov2.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_Cov2_over_Mock_woCov2.pdf")
```

#Interpretation

Gene set annotations are downloaded from MSigDB
https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp#H
29.05.2020: 14:42 CET
C5 : GO Bioprocess

C7 : Immunology

h.all : Hallmark gene sets

Citation for hallmark collection:

Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi M, Mesirov JP, Tamayo P. The Molecular Signatures Database (MSigDB) hallmark gene set collection. Cell Syst. 2015 Dec 23;1(6):417-425. 
 


## Example collection

Next we plot the top differential expressed genes as a function of time and infection

```{r}

LUT <- star_data_id2name %>% column_to_rownames("Geneid")

topGenes_up <- xkira_over_Mock %>%
  as.data.frame() %>%
  arrange(desc(stat)) %>%
  rownames_to_column("name") %>%
  filter(log2FoldChange>0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_up <- map(topGenes_up, function(topGene){#nrow()
   geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("concentration"),returnData = TRUE) %>%  mutate(concentration=factor(concentration,levels = c("0uM","1uM","2uM"),ordered = T))
   ggplot(geneCounts, aes(x = concentration, y = count)) +
              scale_y_log10() +  
              geom_point(size = 3) + 
              geom_smooth() +
              theme_b110() +
              scale_color_manual(values = c(google_red,b110_grey)) + 
              ylab("norm. count") +
              xlab("time [h]") +
              ggtitle(LUT[topGene,1])
})

topGenes_down <- xkira_over_Mock %>%
  as.data.frame() %>%
  arrange(stat) %>%
  rownames_to_column("name") %>%
  filter(log2FoldChange<0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_down <- map(topGenes_down, function(topGene){#nrow()
  geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("concentration"),returnData = TRUE) %>%  mutate(concentration=factor(concentration,levels = c("0uM","1uM","2uM"),ordered = T))
   ggplot(geneCounts, aes(x = concentration, y = count)) +
            scale_y_log10() +  
            geom_point(size = 3) + 
            geom_smooth() +
            theme_b110() +
            scale_color_manual(values = c(google_red,b110_grey)) + 
            ylab("norm. count") +
            xlab("time [h]") +
            ggtitle(LUT[topGene,1])
})

## plot to canvas
a<-reduce(c(count_plots_up), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave(filename = "plots/star_top_up-regulated_genes_xkira_over_Mock.pdf",plot = a,width = 20,height = 20,units = "cm")

## plot to canvas
b<-reduce(c(count_plots_down), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave(filename = "plots/star_top_down-regulated_genes_xkira_over_Mock.pdf",plot = b,width = 20,height = 20,units = "cm")

```

# kira DEG vs Cov2 DEG

```{r}
require(eulerr)
xkira_DEG <- xkira_over_Mock %>% 
   as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)%>% 
  filter(padj<0.1,grepl("ENSG",Geneid))


Cov2_DEG <- Cov2_over_Mock %>% 
   as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name) %>% 
  filter(padj<0.1,grepl("ENSG",Geneid))

joined_DEG <- xkira_DEG %>%
  full_join(Cov2_DEG,suffix = c("kira","Cov2"),by = c("Geneid","gene_name")) %>%
  as_tibble()

joined_DEG %>% drop_na() %>% ggplot(aes(x=log2FoldChangekira,y=log2FoldChangeCov2)) + geom_point() + theme_b110() + geom_vline(xintercept = 0) + geom_hline(yintercept = 0)

joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2>0,log2FoldChangekira>0) %>%
  write_delim("processed_data/Cov2_shared_kira_positive_DEG.txt",delim = "\t")

joined_DEG %>% 
  drop_na() %>%
  filter(log2FoldChangeCov2<0,log2FoldChangekira<0) %>%
  write_delim("processed_data/Cov2_shared_kira_negative_DEG.txt",delim = "\t")


p1 <- euler(list("xkira" = joined_DEG %>% 
  filter(!is.na(log2FoldChangekira),log2FoldChangekira<0) %>% 
  pull(gene_name),
  "Cov2" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeCov2),log2FoldChangeCov2<0) %>% 
  pull(gene_name)))

plot(p1,main="down regulated genes")

pdf("plots/kira_Cov2_shared_downregulated_genes.pdf")
plot(p1,main="down regulated genes")
dev.off()

p2 <- euler(list("xkira" = joined_DEG %>% 
  filter(!is.na(log2FoldChangekira),log2FoldChangekira>0) %>% 
  pull(gene_name) %>% unique(),
  "Cov2" = joined_DEG %>% 
  filter(!is.na(log2FoldChangeCov2),log2FoldChangeCov2>0) %>% 
  pull(gene_name)%>% unique()))

plot(p2,main="up regulated genes")
pdf("plots/kira_Cov2_shared_upregulated_genes.pdf")
plot(p2,main="up regulated genes")
dev.off()
```

# Cov2 genes upon kira treatment

```{r}
# with Cov2 genes
LUT <- star_data_id2name %>% column_to_rownames("Geneid")

topGenes_updown <- Cov2_over_Mock %>%
  as.data.frame() %>%
  arrange(desc(abs(stat))) %>%
  rownames_to_column("name") %>%
  filter(log2FoldChange!=0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_up <- map(topGenes_updown, function(topGene){#nrow()
   geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("concentration","infection"),returnData = TRUE) %>%  mutate(concentration=factor(concentration,levels = c("0uM","1uM","2uM"),ordered = T))
   ggplot(geneCounts, aes(x = concentration, y = count,col=infection)) +
              scale_y_log10() +  
              geom_point(size = 3) + 
              geom_smooth() +
              theme_b110() +
              scale_color_manual(values = c(google_red,b110_grey)) + 
              ylab("norm. count") +
              xlab("time [h]") +
              ggtitle(LUT[topGene,1])
})

a<-reduce(c(count_plots_up), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave(filename = "plots/star_top_Cov2-regulated_genes_xkira_over_Mock.pdf",plot = a,width = 30,height = 20,units = "cm")

# without Cov2 genes
LUT <- star_data_id2name %>% column_to_rownames("Geneid")

topGenes_updown <- Cov2_over_Mock %>%
  as.data.frame() %>%
  arrange(desc(abs(stat))) %>%
  rownames_to_column("name") %>%
  filter(!grepl("ENSSAS",name)) %>%
  filter(log2FoldChange!=0) %>%
  head(n=20) %>%
  pull(name)

## generate plots
count_plots_up <- map(topGenes_updown, function(topGene){#nrow()
   geneCounts <- plotCounts(dds, gene = topGene, intgroup = c("concentration","infection"),returnData = TRUE) %>%  mutate(concentration=factor(concentration,levels = c("0uM","1uM","2uM"),ordered = T))
   ggplot(geneCounts, aes(x = concentration, y = count,col=infection)) +
              scale_y_log10() +  
              geom_point(size = 3) + 
              geom_smooth() +
              theme_b110() +
              scale_color_manual(values = c(google_red,b110_grey)) + 
              ylab("norm. count") +
              xlab("time [h]") +
              ggtitle(LUT[topGene,1])
})

a<-reduce(c(count_plots_up), `+`) + plot_layout(ncol = 5, nrow = 4)

ggsave(filename = "plots/star_top_Cov2-regulated_genes_xkira_over_Mock_woCov2.pdf",plot = a,width = 30,height = 20,units = "cm")
```


# write table of detected genes

```{r}

tibble("Geneid"=rownames(xkira_over_Mock)) %>% left_join(star_data_id2name) %>% write_delim("processed_data/detected_genes_kira_experiment.txt",delim = "\t")

```

# Session info

```{r}
writeLines(capture.output(sessionInfo()), "processed_data/SessionInfo_kira_experiment.txt")
```







