---
title: "RNA_Seq_AnalysisCalu3_Mock_Cov2"
output:
  html_document: default
  pdf_document: default
editor_options: 
  chunk_output_type: console
---

## Setup

First we need to load all sorts of packages that enable us to analyze a read count table that we generated using the nextflow nf-core pipeline version 1.4.3 on a fused genome build of human and SARS-Cov2.

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo =     FALSE,
	message =  FALSE,
	warning =  FALSE
)

library(ggrepel)
library(DESeq2)
library(tidyverse)
library(patchwork)
library(ggrastr)
library(fgsea)

# B110 Theme

theme_b110 <- function(){
  theme_classic() +
  theme(
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 10),
    plot.title = element_text(size = 12,hjust = 0.5,face = "bold"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10),
    legend.position = "bottom"
    )
}

sgi_blue    = '#5087C8'
sgi_yellow1 = '#F2EE35'
sgi_yellow2 = '#FED98E'
b110_grey   = '#808080'
b110_grey_light   = '#909090'
b110_transparent_black = alpha('#000000',0.5)
google_red = '#dd4b39'
google_green = '#0F9D58'
google_yellow = '#F4B400'
google_blue = '#4285F4'

```

## Read in the meta data

```{r meta_data}

raw_meta_data <- read_delim("meta_data/31983_meta.tsv",delim = "\t")

raw_meta_data %<>% 
  filter(grepl("R1",FASTQ_FILE)) %>% 
  mutate(
    run = gsub(pattern = "(.+)_.*.fastq.gz",
               replacement = "\\1",x = FASTQ_FILE)) %>% 
  dplyr::rename(sample = SAMPLE_NAME)

raw_meta_data_1 <-
    raw_meta_data %>% separate(sample,c("number","infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  filter(!grepl("KIRA",sample)) %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  mutate(treatment=if_else(treatment!="NonInf","4u8C",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()
  
raw_meta_data_2 <-
    raw_meta_data %>% separate(sample,c("number","infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  filter(treatment!="4µ8C") %>%
   filter(treatment!="4u8C") %>%
  mutate(treatment=if_else(treatment!="NonInf","Kira",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()

raw_meta_data <- bind_rows(raw_meta_data_1,raw_meta_data_2)
```

#import the data from star

```{r,warning=F,message=F}

star_data <- read_delim("data/merged_gene_counts.txt",delim = "\t")

colnames(star_data) <- gsub(pattern = "(\\S+?)Aligned.sortedByCoord.out.bam",replacement = "\\1",x = colnames(star_data))

star_data_id2name <- star_data %>% select(-starts_with("AS"))

star_data %<>%
  gather(run,count,-Geneid,-gene_name) %>% 
  separate(run,c("AS","number1","LR","number2")) %>%
  group_by(Geneid,gene_name,AS,number1) %>%
  summarise(count=sum(count)) %>%
  ungroup() %>% 
  select(Geneid,gene_name,number1,count)

#star_data %<>% 
#  select(-gene_name) %>% 
#  column_to_rownames("Geneid") %>% 
#  as.data.frame()

#star_data <- star_data[-grep("ENSSAS",rownames(star_data)),]

```

#prepare the data object

```{r,warning=F,message=F}

# filter out genes with low expression or coverage to this end, we used a reasonable threshold of 10 reads averaged across all samples

star_data_filtered <- 
  star_data %>% 
  group_by(Geneid) %>%
  filter(mean(count)>10) %>%
  ungroup()

# this filter leaves us with 16,293 expressed genes 
# next we normalize the data by each samples median, variance-mean relation and apply a rlog transform

star_data_normed <- 
  star_data_filtered %>% 
  select(-gene_name) %>% 
  spread(number1,count) %>%
  column_to_rownames("Geneid") %>% 
  as.matrix() %>%
  varianceStabilizingTransformation(object = .) %>%
  as.data.frame() %>%
  rownames_to_column("Geneid") %>%
  left_join(star_data_id2name) %>%
  tibble() %>%
  gather(run,norm_expr,-Geneid,-gene_name)

# next we bring normalized data and annotations together

star_data_annotated <- 
  star_data_normed %>%
    left_join(raw_meta_data) %>%
    mutate(treatment = if_else(concentration =="0uM", "DMSO",treatment)) %>%
    distinct()

star_data_foldchange_dmso <- 
star_data_annotated %>%
  select(-sample,-run) %>%
  unite(treatment,concentration,treatment) %>%
  spread(treatment,norm_expr) %>%
  mutate(across(one_of(c("100uM_4u8C","1uM_Kira","2uM_Kira","50uM_4u8C")),~.x-`0uM_DMSO`)) %>%
  select(-`0uM_DMSO`) %>%
  gather(treatment,log_fc,-Geneid,-gene_name,-infection) %>%
  separate(treatment,c("concentration","treatment"))

```

# Viral load analysis

```{r}

a <- star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = gene_name, fill = `foldchange Drug/DMSO [log]`)) +
    geom_tile() +
    scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110() +
    xlab("concentration [µM]")+
    ylab("Gene")

b <- star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
   rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = `foldchange Drug/DMSO [log]`, group=gene_name,col=viral_gene)) +
    geom_line() +
    #scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110()

star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
   rename(`foldchange Drug/DMSO [log]`=log_fc) %>% write_delim("processed_data/viral_reads_drug_effect.csv",delim = ",")

a+b

ggsave(filename = "plots/viral_gene_Kira_4µ8c_effect4.pdf",a+b,w=14,height=9,units = "cm")


gene_name_ordered_100uM <- 
    star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  filter(concentration=="100uM") %>%
  arrange(log_fc) %>%
  pull(gene_name) %>%
  unique()

c <- star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = factor(gene_name,levels = gene_name_ordered_100uM), fill = `foldchange Drug/DMSO [log]`)) +
    geom_tile() +
    scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110() +
    xlab("concentration [µM]")+
    ylab("Gene") + 
    theme(
      axis.text.y = element_blank(),
      axis.ticks = element_blank())

star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>% write_delim("processed_data/full_reads_drug_effect.csv",delim = ",")


ggsave(filename = "plots/gene_Kira_4µ8c_effect_large.pdf",c,w=14,height=20,units = "cm")
    


d <- star_data_foldchange_dmso  %>% 
  filter(infection=="NonInf") %>% 
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  ggplot(aes(x=log_fc,y=mean_expr,col=viral_gene)) +
    geom_point() +
    facet_wrap(~treatment+concentration) +
      theme_gray() +
    xlab("foldchange Drug/DMSO [log]")+
    ylab("Avg. expression [AU]") +
    ggtitle("Expression vs. Foldchange (non-infected)")

ggsave(filename = "plots/expression_Kira_4µ8c_effect_volcano_nonInf.pdf",d,w=20,height=20,units = "cm")
 

e <- star_data_foldchange_dmso  %>% 
  filter(infection=="MOI10")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  ggplot(aes(x=log_fc,y=mean_expr,col=viral_gene)) +
    geom_point() +
    facet_wrap(~treatment+concentration)+
    theme_gray() +
    xlab("foldchange Drug/DMSO [log]")+
    ylab("Avg. expression [AU]") +
    ggtitle("Expression vs. Foldchange (Cov2-infected)")

star_data_foldchange_dmso  %>% 
  filter(infection=="MOI10")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>% write_delim("processed_data/expression_vs_Kira_4µ8c_effect_Cov2Inf.csv",delim = ",")

star_data_foldchange_dmso  %>% 
  filter(infection=="NonInf")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>% write_delim("processed_data/expression_vs_drug_effect_Non2Inf.csv",delim = ",")


ggsave(filename = "plots/expression_Kira_4µ8c_effect_volcano_Cov2Inf.pdf",e,w=20,height=20,units = "cm")
 
  



```



# write table of detected genes

```{r}

tibble("Geneid"=star_data_filtered$Geneid) %>% left_join(star_data_id2name) %>% distinct() %>% write_delim("processed_data/detected_genes_Kira_4µ8c_experiment.txt",delim = "\t")

```

# Session info

```{r}
writeLines(capture.output(sessionInfo()), "processed_data/SessionInfo_Kira_4µ8c_experiment.txt")
```
