---
title: "RNA_Seq_AnalysisCalu3_Mock_Cov2"
output:
  html_document: default
  pdf_document: default
editor_options: 
  chunk_output_type: console
---

## Setup

First we need to load all sorts of packages that enable us to analyze a read count table that we generated using the nextflow nf-core pipeline version 1.4.3 on a fused genome build of human and SARS-Cov2.

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo =     FALSE,
	message =  FALSE,
	warning =  FALSE
)

library(ggrepel)
library(DESeq2)
library(tidyverse)
library(patchwork)
library(ggrastr)
library(fgsea)

# B110 Theme

theme_b110 <- function(){
  theme_classic() +
  theme(
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 10),
    plot.title = element_text(size = 12,hjust = 0.5,face = "bold"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10),
    legend.position = "bottom"
    )
}

sgi_blue    = '#5087C8'
sgi_yellow1 = '#F2EE35'
sgi_yellow2 = '#FED98E'
b110_grey   = '#808080'
b110_grey_light   = '#909090'
b110_transparent_black = alpha('#000000',0.5)
google_red = '#dd4b39'
google_green = '#0F9D58'
google_yellow = '#F4B400'
google_blue = '#4285F4'

```

## Read in the meta data

```{r meta_data}

raw_meta_data <- read_delim("meta_data/31983_meta.tsv",delim = "\t")

raw_meta_data2 <- read_delim("meta_data/32516_meta.tsv",delim = "\t")

raw_meta_data %<>% 
  filter(grepl("R1",FASTQ_FILE)) %>% 
  mutate(
    run = gsub(pattern = "(.+)_.*.fastq.gz",
               replacement = "\\1",x = FASTQ_FILE)) %>% 
  dplyr::rename(sample = SAMPLE_NAME)

raw_meta_data2 %<>% 
  filter(grepl("R1",FASTQ_FILE)) %>% 
  mutate(
    run = gsub(pattern = "(.+)_.*.fastq.gz",
               replacement = "\\1",x = FASTQ_FILE)) %>% 
  dplyr::rename(sample = SAMPLE_NAME)

raw_meta_data_1 <-
    raw_meta_data %>% separate(sample,c("number","infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  filter(!grepl("KIRA",sample)) %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  mutate(treatment=if_else(treatment!="NonInf","4u8C",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()
  
raw_meta_data_2 <-
    raw_meta_data %>% separate(sample,c("number","infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  filter(treatment!="4µ8C") %>%
   filter(treatment!="4u8C") %>%
  mutate(treatment=if_else(treatment!="NonInf","Kira",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()

raw_meta_data2_1 <-
    raw_meta_data2 %>% separate(sample,c("infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  filter(!grepl("Kira",sample)) %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  mutate(treatment=if_else(treatment!="NonInf","4u8C",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()
  
raw_meta_data2_2 <-
     raw_meta_data2 %>% separate(sample,c("infection","treatment","concentration"),remove = F,sep = "[|\ |\\-|_]") %>%
  select(run,sample,infection,treatment,concentration) %>%
  distinct() %>%
  extract(run,"run","AS-(\\d+)-.+") %>%
  filter(treatment!="4µ8C") %>%
   filter(treatment!="4u8C") %>%
  mutate(treatment=if_else(treatment!="NonInf","Kira",treatment),
         concentration=factor(if_else(is.na(concentration),"0µM",concentration))) %>%
  mutate(concentration=gsub("µ","u",concentration)) %>%
  distinct()

raw_meta_data <- bind_rows(raw_meta_data_1,raw_meta_data_2,raw_meta_data2_1,raw_meta_data2_2)
```

#import the data from star

```{r,warning=F,message=F}

star_data <- read_delim("data/merged_gene_counts_2.txt",delim = "\t")

colnames(star_data) <- gsub(pattern = "(\\S+?)Aligned.sortedByCoord.out.bam",replacement = "\\1",x = colnames(star_data))

star_data_id2name <- star_data %>% select(-starts_with("AS"))

star_data %<>%
  gather(run,count,-Geneid,-gene_name) %>% 
  separate(run,c("AS","number1","LR","number2")) %>%
  group_by(Geneid,gene_name,AS,number1) %>%
  summarise(count=sum(count)) %>%
  ungroup() %>% 
  select(Geneid,gene_name,number1,count)

star_data_deseq <- star_data %>% 
  select(-gene_name) %>% 
  spread(number1,count) %>% 
  column_to_rownames("Geneid") %>%
  as.data.frame()

#star_data <- star_data[-grep("ENSSAS",rownames(star_data)),]

```

#prepare the data object

```{r,warning=F,message=F}

# filter out genes with low expression or coverage to this end, we used a reasonable threshold of 10 reads averaged across all samples

star_data_filtered <- 
  star_data %>% 
  group_by(Geneid) %>%
  filter(mean(count)>10) %>%
  ungroup()

# this filter leaves us with 16,293 expressed genes 
# next we normalize the data by each samples median, variance-mean relation and apply a rlog transform

star_data_normed <- 
  star_data_filtered %>% 
  select(-gene_name) %>% 
  spread(number1,count) %>%
  column_to_rownames("Geneid") %>% 
  as.matrix() %>%
  varianceStabilizingTransformation(object = .) %>%
  as.data.frame() %>%
  rownames_to_column("Geneid") %>%
  left_join(star_data_id2name) %>%
  tibble() %>%
  gather(run,norm_expr,-Geneid,-gene_name)

# next we bring normalized data and annotations together

star_data_annotated <- 
  star_data_normed %>%
    left_join(raw_meta_data) %>%
    mutate(treatment = if_else(concentration =="0uM", "DMSO",treatment)) %>%
    distinct()

star_data_foldchange_dmso <- 
star_data_annotated %>%
  group_by(Geneid,gene_name,infection,treatment,concentration) %>%
  summarise(norm_expr=mean(norm_expr)) %>%
  ungroup() %>%
  unite(treatment,concentration,treatment) %>%
  spread(treatment,norm_expr) %>%
  mutate(across(one_of(c("100uM_4u8C","1uM_Kira","2uM_Kira","50uM_4u8C")),~.x-`0uM_DMSO`)) %>%
  select(-`0uM_DMSO`) %>%
  gather(treatment,log_fc,-Geneid,-gene_name,-infection) %>%
  separate(treatment,c("concentration","treatment"))

```

# Viral load analysis

```{r}

a <- star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = gene_name, fill = `foldchange Drug/DMSO [log]`)) +
    geom_tile() +
    scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110() +
    xlab("concentration [µM]")+
    ylab("Gene")

b <- star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
   rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = `foldchange Drug/DMSO [log]`, group=gene_name,col=viral_gene)) +
    geom_line() +
    #scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110()

star_data_foldchange_dmso %>%
  filter(grepl("ENSSAS",Geneid) | gene_name %in% c("GAPDH","IDH1","IDH2"),infection=="MOI10") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  mutate(gene_name=factor(gene_name,levels = c("GAPDH","IDH1","IDH2","ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  mutate(viral_gene=!(gene_name %in% c("GAPDH","IDH1","IDH2"))) %>%
   rename(`foldchange Drug/DMSO [log]`=log_fc) %>% write_delim("processed_data/viral_reads_drug_effect.csv",delim = ",")

a+b

ggsave(filename = "plots/viral_gene_Kira_4µ8c_effect4.pdf",a+b,w=14,height=9,units = "cm")


gene_name_ordered_100uM <- 
    star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  filter(concentration=="100uM") %>%
  arrange(log_fc) %>%
  pull(gene_name) %>%
  unique()

c <- star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>%
  ggplot(aes(x = concentration, y = factor(gene_name,levels = gene_name_ordered_100uM), fill = `foldchange Drug/DMSO [log]`)) +
    geom_tile() +
    scale_fill_gradient2() +
    facet_wrap(~treatment,scales = "free_x") +
    theme_b110() +
    xlab("concentration [µM]")+
    ylab("Gene") + 
    theme(
      axis.text.y = element_blank(),
      axis.ticks = element_blank())

star_data_foldchange_dmso %>%
  filter(infection=="NonInf") %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  rename(`foldchange Drug/DMSO [log]`=log_fc) %>% write_delim("processed_data/full_reads_drug_effect.csv",delim = ",")


ggsave(filename = "plots/gene_Kira_4µ8c_effect_large.pdf",c,w=14,height=20,units = "cm")
    


d <- star_data_foldchange_dmso  %>% 
  filter(infection=="NonInf") %>% 
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  ggplot(aes(x=log_fc,y=mean_expr,col=viral_gene)) +
    geom_point() +
    facet_wrap(~treatment+concentration) +
      theme_gray() +
    xlab("foldchange Drug/DMSO [log]")+
    ylab("Avg. expression [AU]") +
    ggtitle("Expression vs. Foldchange (non-infected)")

ggsave(filename = "plots/expression_Kira_4µ8c_effect_volcano_nonInf.pdf",d,w=20,height=20,units = "cm")
 

e <- star_data_foldchange_dmso  %>% 
  filter(infection=="MOI10")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>%
  ggplot(aes(x=log_fc,y=mean_expr,col=viral_gene)) +
    geom_point() +
    facet_wrap(~treatment+concentration)+
    theme_gray() +
    xlab("foldchange Drug/DMSO [log]")+
    ylab("Avg. expression [AU]") +
    ggtitle("Expression vs. Foldchange (Cov2-infected)")

star_data_foldchange_dmso  %>% 
  filter(infection=="MOI10")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>% write_delim("processed_data/expression_vs_Kira_4µ8c_effect_Cov2Inf.csv",delim = ",")

star_data_foldchange_dmso  %>% 
  filter(infection=="NonInf")  %>%
  mutate(concentration=factor(concentration,levels = c("1uM","2uM","50uM","100uM"))) %>%
  left_join(star_data_annotated %>%
    select(Geneid,norm_expr) %>%
    group_by(Geneid) %>%
    summarise(mean_expr=mean(norm_expr,na.rm=T)) %>%
    ungroup()
    ) %>%
  mutate(viral_gene=(gene_name %in% c("ORF1ab","S","N","ORF3a","M","ORF8","ORF7a","E","ORF6","ORF7b","ORF10"))) %>% write_delim("processed_data/expression_vs_drug_effect_Non2Inf.csv",delim = ",")


ggsave(filename = "plots/expression_Kira_4µ8c_effect_volcano_Cov2Inf.pdf",e,w=20,height=20,units = "cm")
 
  



```


## Analysis for drug testing DESeq2

# design table preparation

```{r}

sampleTable_drug_experiment <- 
  raw_meta_data %>% 
    filter(concentration %in% c("2uM","100uM","0uM"))  %>% #,"100uM"#
   # mutate(treatment=if_else(treatment=="4µ8C","4u8C",treatment)) %>%
    mutate(treatment = if_else(concentration=='0uM','DMSO',treatment),infection=factor(infection),treatment=factor(treatment)) %>% 
    select(-sample,-concentration) %>%
    distinct() %>%
   
    as_data_frame()

rownames(sampleTable_drug_experiment)<-sampleTable_drug_experiment$run

```


#simple prefiltering

```{r,warning=F,message=F}

star_data_drug_experiment <- star_data_deseq[,rownames(sampleTable_drug_experiment)]

#star_data_drug_experiment <- star_data_drug_experiment[-grep("ENSSAS",rownames(star_data_drug_experiment)),]

dds_drug_experiment <- DESeqDataSetFromMatrix(star_data_drug_experiment, sampleTable_drug_experiment, ~infection+treatment+treatment:infection)

#counting transcripts

nrow(dds_drug_experiment)

#excluding genes that were not matched at all
dds_drug_experiment <- dds_drug_experiment[ rowSums(counts(dds_drug_experiment)) > 10, ]
nrow(dds_drug_experiment)

#remaining 22316

```

#stabilize variance for low readcount genes

```{r,warning=F,message=F}

vsd <- vst(dds_drug_experiment, blind = FALSE)

```

#cluster data to get an overview

Treatments should cluster together

```{r,warning=F,message=F}
require("pheatmap")
require("RColorBrewer")

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- vsd$run
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
annotations = data.frame("virus"=vsd$infection)
row.names(annotations)<-vsd$run

pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,annotation_row = annotations)

c <- plotPCA(vsd, c("infection")) + ggtitle("PCA by virus") + theme_b110()
d <- plotPCA(vsd, c("treatment")) + ggtitle("PCA by drug")+ theme_b110()

print(c+d)
ggsave(c+d,filename = "plots/PCA_pre_batch_correct_dds_drug_experiment_wCov2.pdf")

```

# lets call differential genes

```{r,warning=F,message=F}

dds <- DESeq(dds_drug_experiment)

# get available contrasts

resultsNames(dds)

# choose the contrast which shows the difference between treatments given background differences of the treatment response over time.

x4u8C_over_Mock <- results(dds,contrast = c("treatment","4u8C","DMSO"))

xkira_over_Mock <- results(dds,contrast = c("treatment","Kira","DMSO"))

xkira_over_4u8C <- results(dds,contrast = c("treatment","Kira","4u8C"))

Cov2_over_Mock <- results(dds,contrast = c("infection","MOI10","NonInf"))

interaction_Cov2_over_Mock_DMSO <- results(dds,name = "infectionNonInf.treatmentDMSO")
interaction_Cov2_over_Mock_Kira <- results(dds,name = "infectionNonInf.treatmentKira")

summary(interaction_Cov2_over_Mock_DMSO)
summary(interaction_Cov2_over_Mock_Kira)

summary(x4u8C_over_Mock)
summary(xkira_over_Mock)
summary(xkira_over_4u8C)
summary(Cov2_over_Mock)

#res %>% as.data.frame() %>% View()
# --> 4µ8C changes the transcriptome a lot (> 2000 DEG)
# --> Kira changes the transcriptome very mildly (<30 DEG)
# --> Infection changes the transcriptome very little (<3 DEG (disregarding viral transcripts))
# --> Not transcript thats changed upon infection is significantly changed upon treatment (<1 DEG (disregarding viral transcripts))


```



# Cov2 vs drug

```{r,warning=F,message=F}

plotCounts(dds, gene = "ENSG00000120738", intgroup = c('treatment',"infection"),returnData = F) 

# here we threshold the data simply in upregulated and down regulated according to foldchange and sort by FDR
# for enrichment of functional annotation clustering it make further sense to threshold by fold change and FDR

xkira_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>%
  arrange(padj) %>% 
  write_delim(path = "processed_data/star_top_table_xkira_over_mock_full.txt",delim = "\t")

down_regulated_genes_drug <- xkira_over_Mock %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>% 
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange<0,padj<0.1)
  
write_delim(down_regulated_genes_drug,"processed_data/star_top_table_xkira_over_Mock_downdrug.txt",delim = "\t")

up_regulated_genes_drug <- xkira_over_Mock %>% 
  as.data.frame() %>%
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name)  %>% 
  arrange(padj) %>% 
  filter(log2FoldChange>0,padj<0.1)

write_delim(up_regulated_genes_drug,"processed_data/star_top_table_xkira_over_Mock_updrug.txt",delim = "\t")


```

##Vizual inspection

#xdrug_over_Mock

```{r}
# next we visualize the differential expressed genes using MA plot of base line expression versus fold change and foldchange versus -log10 pvalue
# here all genes that are deferentially expressed between SARS-Cov2 infection and Mock treatment with an FDR < 0.1 are colored blue.
# the top 15 differentially expressed genes according the the FDR ranking are labeled by their official gene symbol

this_data <- xkira_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na()

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_xkira_over_Mock.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_xkira_over_Mock.pdf")

# next we visualize the same data without the viral genes


this_data <- xkira_over_Mock  %>%
  as.data.frame() %>% 
  rownames_to_column(var = "Geneid") %>%  
  left_join(star_data_id2name) %>% 
  mutate(significant = if_else(padj<=0.1,"significant","non-significant")) %>%
  drop_na() %>%
  filter(grepl("ENSG",Geneid))

 p1 <-this_data %>%
         ggplot(aes(x=log2FoldChange,y=-log10(pvalue),col=significant,key=Geneid,gene_id=gene_name))  +
            geom_point(data = this_data %>% filter(significant=="non-significant"),alpha=0.3) +
            geom_point(data = this_data %>% filter(significant!="non-significant")) +
            geom_vline(xintercept = 0,lty=2,col=b110_grey_light) +
           geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,30),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
            theme_b110() +
            scale_color_manual(values = c(b110_grey_light,google_blue))+ 
            ggtitle("Treatment differential Volcano-plot")


p2 <-this_data %>%
  ggplot(aes(x=baseMean,y=log2FoldChange,col=significant,key=Geneid,gene_id=gene_name)) +
    geom_point(data = subset(this_data,significant=="non-significant"),alpha=0.3) +
    geom_smooth(col="black") +
    geom_point(data = subset(this_data,significant=="significant")) +
     geom_label_repel(data = this_data %>% filter(significant=="significant") %>% arrange(padj) %>% head(.,15),aes(label=gene_name),col=b110_grey,min.segment.length = 0) +
    theme_b110() +
    scale_color_manual(values = c(b110_grey_light,google_blue)) + 
    ggtitle("Treatment differential MA-plot") +
    scale_x_log10()

ggsave(p1,filename = "plots/star_differential_expressed_volcano_xkira_over_Mock_woCov2.pdf")
ggsave(p2,filename = "plots/star_differential_expressed_MA_xkira_over_Mock_woCov2.pdf")
```


# write table of detected genes

```{r}

tibble("Geneid"=star_data_filtered$Geneid) %>% left_join(star_data_id2name) %>% distinct() %>% write_delim("processed_data/detected_genes_Kira_4µ8c_experiment.txt",delim = "\t")

```

# Session info

```{r}
writeLines(capture.output(sessionInfo()), "processed_data/SessionInfo_Kira_4µ8c_experiment.txt")
```
